lightning/src/util/test_utils.rs

```diff
@@ -93,13 +93,16 @@ impl<'a> chain::Watch for TestChainMonitor<'a> {
 		assert!(new_monitor == monitor);
 		self.latest_monitor_update_id.lock().unwrap().insert(funding_txo.to_channel_id(), (funding_txo, monitor.get_latest_update_id()));
 		self.added_monitors.lock().unwrap().push((funding_txo, monitor));
-		assert!(self.chain_monitor.watch_channel(funding_txo, new_monitor).is_ok());
+		let watch_res = self.chain_monitor.watch_channel(funding_txo, new_monitor);
 
 		let ret = self.update_ret.lock().unwrap().clone();
 		if let Some(next_ret) = self.next_update_ret.lock().unwrap().take() {
-			*self.update_ret.lock().unwrap() = next_ret;
+			*self.update_ret.lock().unwrap() = Some(next_ret);
 		}
-		ret
+		if ret.is_some() {
+			return ret.clone().unwrap();
```

Comment:
Mostly as a sanity check for the tests, should we assert that `watch_res` is ok here? Likewise below.